drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/DecisionTableAnalyzer.java

```diff
@@ -146,19 +141,28 @@ protected AnalysisReport makeAnalysisReport() {
         return report;
     }
 
-    protected void sendReport( final AnalysisReport report ) {
-        IOC.getBeanManager().lookupBean( AnalysisReportScreen.class ).getInstance().showReport( report );
+    private void sendReport( final AnalysisReport report ) {
+        getAnalysisReportScreen().showReport( report );
+    }
+
+    private void sendStatus( final Status status ) {
+        getAnalysisReportScreen().showStatus( status );
+    }
+
+    protected AnalysisReportScreen getAnalysisReportScreen() {
+        return IOC.getBeanManager().lookupBean( AnalysisReportScreen.class ).getInstance();
```

Comment:
I know this existed in the old code; but if we're instantiating a managed bean programmatically we need also to handle the life-cycle (i.e. destroy it when done). It might be better to ```@Inject``` ```AnalysisReportScreen``` into ```GuidedDecisionTablePresenter``` and pass as a parameter to ```DecisionTableAnalyzer``` constructor. 

Just a thought.. otherwise we should probably only create a single instance in ```DecisionTableAnalyzer``` constructor and destroy it when the editor closes. 

```AnalysisReportScreenActivity``` is ```@Dependent``` meaning every time an instance is requested a new one is created... so ```getAnalysisReportScreen()``` is probably creating a new instance each time it's called.